### DISCUSSION TITLE
What's the `eof` attribute?

### DISCUSSION THREAD

**Discussion by EliahKagan** [GitoxideLabs/gitoxide#1689] [Q&A] [answered]
One of the `.gitattributes` files in the gitoxide repo unsets the `eof` attribute on some files containing test data:

https://github.com/GitoxideLabs/gitoxide/blob/b06f7299cf0fd75c878a3e61a80b362bccb16943/gix-config/tests/fixtures/fuzzed/.gitattributes#L1

I have not been able to find any documentation on the `eof` attribute. I don't think Git itself recognizes or uses it, but maybe some other tool does.

Does this prevent some tool from automatically adding/removing line endings at the very end of files? I don't think Git ever adds them automatically (it will convert between LF and CRLF if so configured, but not, as far as I know, change the number of line endings). Other tools that use a repository's Git attributes can recognize their own, but it would be very strange if any tool automatically normalized line endings in files for which `text` has been unset, as is also done there. So I'm not sure what `-eof` would do, even hypothetically.

The file was created with those contents in 6bb407f, along with the other contents of the `fixtures/fuzzed` directory. Examining the context there doesn't seem to reveal the effect or intent of `-eof`.

**Comment by Byron** [comment, marked as answer]
I am sorry for that. No mention of this ominous attribute probably means that it's a mistake. It was probably meant to be `eol`, which does exist but still would be superfluous here and/or is used incorrectly.

**Reply by EliahKagan** [reply]
Thanks. I plan to open a PR in a few minutes to fix an error in `.gitattributes` I inadvertently introduced in c957ab8 (#1340), where the actual files inside `gix-packetline-blocking/src` still had `linguist-vendored` unspecified, and also simplifies `.gitattributes` files throughout the project. I'll include the removal of this `-eof` in that PR as well.

*Edit:* I've opened #1690, which includes this.
